src/viur/core/i18n.py

```diff
@@ -1,21 +1,109 @@
+    # that should be available for the project.
+    conf.i18n.available_languages = = ["en", "de", "fr"]
+
+    # These are some aliases for languages that should use the translated
+    # values of a particular main language, but don't have their own values.
+    conf.i18n.language_alias_map = {
+        "at": "de",  # Austria uses German
+        "ch": "de",  # Switzerland uses German
+        "be": "fr",  # Belgian uses France
+        "us": "en",  # US uses English
+    }
+
+Now translations can be used
+
+1. In python
+.. code-block:: python
+    from viur.core.i18n import translate
+    # Just the translation key, the minimal case
+    print(translate("translation-key"))
+    # also provide a default value to use if there's no value in the datastore
+    # set and a hint to provide some context.
+    print(translate("translation-key", "the default value", "a hint"))
+    # Use string interpolation with variables
+    print(translate("hello", "Hello {{name}}!", "greeting a user")(name=current.user.get()["firstname"]))
+
+2. In jinja
+.. code-block:: jinja
+    {# Use the ViUR translation extension, it can be compiled with the template,
+       caches the translation values and is therefore efficient #}
+    {% do translate "hello", "Hello {{name}}!", "greet a user", name="ViUR" %}
+
+   {# But in some cases the key or interpolation variables are dynamic and
+      not available during template compilation.
+      For this you can use the translate function: #}
+    {{ translate("hello", "Hello {{name}}!", "greet a user", name=skel["firstname"]) }}
+
+
+How to add translations
+There are two ways to add translations:
+1. Manually
+With the vi-admin. Entries can be added manually by creating a new skeleton
+and filling in of the key and values.
+
+2. Automatically
+The add_missing_translations option must be enabled for this.
+.. code-block:: python
+
+    from viur.core.config import conf
+    conf.i18n.add_missing_translations = True
+
+
+If a translation is now printed and the key is unknown (because someone has
+just added the related print code), an entry is added in the datastore kind.
+In addition, the default text and the hint are filled in and the filename
+and the line from the call from the code are set in the skeleton.
+This is the recommended way, as ViUR collects all the information you need
+and you only have to enter the translated values.
+(3. own way
+Of course you can create skeletons / entries in the datastore in your project
+on your own. Just use the TranslateSkel).
+"""  # FIXME: grammar, rst syntax
 import datetime
 import jinja2.ext as jinja2
+import logging
+import traceback
 import typing as t
+from pathlib import Path
+
+from viur.core import current, db, languages, tasks
 from viur.core.config import conf
-from viur.core import db, utils, languages, current
 
 systemTranslations = {}
+"""Memory storage for translation methods"""
+
+KINDNAME = "viur-translations"
```

Comment:
I would like to prefer singular names `viur-translation` in this case.
Yes I know, that other modules use plural, but we should start at a point.
```suggestion
KINDNAME = "viur-translation"
```

Reply:
In general, I agree with you. But that would mean that existing data would have to be copied to the new kind.  I would therefore not want to change that.

Or we could change it and then recommend monkeypatching the value for existing projects that already have translations in the DB:
```py
viur.core.i18n.KINDNAME = "viur-translations"
```
But we should test this first to see if it works in time during warmup.

Reply:
Oh... okay then miss this out.